File: Algorithms/p145_Binary_Tree_Postorder_Traversal_1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> res;
        postorderTraversalStep(root,res);
        return res;
    }
    void postorderTraversalStep(TreeNode* root, vector<int>& res) {
        if(root){
            postorderTraversalStep(root->left,res);
            postorderTraversalStep(root->right,res);
            res.push_back(root->val);
        }
    }
};

File: Algorithms/p145_Binary_Tree_Postorder_Traversal_1.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def postorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        res = []
        self.postorderTraversalStep(root,res)
        return res
        
    def postorderTraversalStep(self, root, l):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        if root:
            self.postorderTraversalStep(root.left,l)
            self.postorderTraversalStep(root.right,l)
            l.append(root.val)
